File: native/mixdog-graph/tests/fixtures/calls/go/sample.go
package sample

// hidden()
var quoted = "hidden()"

func inner() {}

func leaf(x int) int { return x }

func nest(x int) {}

func helper() {}

func seed() {}

func plain() int { return 0 }

type Chain struct{}

func (Chain) b() Chain { return Chain{} }

func (Chain) c() Chain { return Chain{} }

var g = plain()

func run(a Chain) {
	inner()
	nest(leaf(1))
	a.b().c()
}

type Widget struct{}

func (Widget) ping() {}

func (w Widget) act() {
	helper()
	w.ping()
}

func seedLine() {
	mark := "μ"; seed()
	_ = new(Widget)
}

File: native/mixdog-graph/tests/fixtures/calls/cpp/sample.cpp
// hidden()
static const char *quoted = "hidden()";

int inner() { return 0; }
int leaf(int x) { return x; }
int nest(int x) { return x; }
int helper() { return 0; }
int seed() { return 0; }
int plain() { return 0; }

struct Chain {
  Chain b() { return *this; }
  Chain c() { return *this; }
};

int g = plain();

void run(Chain a) {
  inner();
  nest(leaf(1));
  a.b().c();
}

class Widget {
public:
  void ping() {}
  void act() {
    helper();
    this->ping();
  }
};

int seed_line() {
  const char *mark = "μ"; seed();
  auto *w = new Widget();
  (void)w;
  return 0;
}
